Makefile

```diff
@@ -1,2 +1,151 @@
+
+is_barrier_hit_test.o: test/is_barrier_hit_test.c
+	$(CC) -g -c $(LIBFLAGS) test/is_barrier_hit_test.c -o build/is_barrier_hit.o
+
+is_player_hit_test.o: test/is_player_hit_test.c
+	$(CC) -g -c $(LIBFLAGS) test/is_player_hit_test.c -o build/is_player_hit_test.o
+
+is_reach_goal_test.o: test/is_reach_goal_test.c
+	$(CC) -g -c $(LIBFLAGS) test/is_reach_goal_test.c -o build/is_reach_goal_test.o
+
+load_map_test.o: test/load_map_test.c
+	$(CC) -g -c $(LIBFLAGS) test/load_map_test.c -o build/load_map_test.o
+
+print_menu_test.o: test/print_menu_test.c
+	$(CC) -g -c $(LIBFLAGS) test/print_menu_test.c -o build/print_menu_test.o
+
+take_heart_test.o: test/take_heart_test.c
+	$(CC) -g -c $(LIBFLAGS) test/take_heart_test.c -o build/take_heart_test.o
+
+update_arrow_test.o: test/update_arrow_test.c
+	$(CC) -g -c $(LIBFLAGS) test/update_arrow_test.c -o build/update_arrow.o
+
+update_barrier_test.o: test/update_barrier_test.c
+	$(CC) -g -c $(LIBFLAGS) test/update_barrier_test.c -o build/update_barrier_test.o
+
+update_player_pos_test.o: test/update_player_pos_test.c
+	$(CC) -g -c $(LIBFLAGS) test/update_player_pos_test.c -o build/update_player_pos_test.o
+
+update_selector_test.o: test/update_selector_test.c
+	$(CC) -g -c $(LIBFLAGS) test/update_selector_test.c -o build/update_selector_test.o
+	
+win_game_test.o: test/win_game_test.c
+	$(CC) -g -c $(LIBFLAGS) test/win_game_test.c -o build/win_game_test.o
+	
+
+
+
+#TARGET TO GENERATE THE EXECUTABLE OF THE PROGRAM (and the tests) - LINK OBJECT FILES
+
+tests: choose_menu_item_test.o game_over_test.o generate_view_test.o is_barrier_hit_test.o is_player_hit_test.o is_reach_goal_test.o load_map_test.o print_menu_test.o take_heart_test.o update_arrow_test.o update_barrier_test.o update_player_pos_test.o update_selector_test.o win_game_test.o
+
+	$(CC) -g -o bin/choose_menu_item_test build/choose_menu_item_test.o
+	$(CC) -g -o bin/game_over_test build/game_over_test.o
+	$(CC) -g -o bin/generate_view_test build/generate_view_test.o
+	$(CC) -g -o bin/is_barrier_hit_test build/is_barrier_hit_test.o
+	$(CC) -g -o bin/is_player_hit_test build/is_player_hit_test.o
+	$(CC) -g -o bin/is_reach_goal_test build/is_reach_goal_test.o
+	$(CC) -g -o bin/load_map_test build/load_map_test.o
+	$(CC) -g -o bin/print_menu_test build/print_menu_test.o
+	$(CC) -g -o bin/take_heart_test build/take_heart_test.o
+	$(CC) -g -o bin/update_arrow_test build/update_arrow_test.o
+	$(CC) -g -o bin/update_barrier_test build/update_barrier_test.o
+	$(CC) -g -o bin/update_player_pos_test build/update_player_pos_test.o
+	$(CC) -g -o bin/update_selector_test build/update_selector_test.o
+	$(CC) -g -o bin/win_game_test build/win_game_test.o
+
+
+	
+main: main.o credit.o game_over.o generate_view.o is_barrier_hit.o is_player_hit.o is_reach_goal.o load_map.o print_menu.o single_player.o take_heart.o update_arrow.o update_barrier.o update_player_pos.o user_manual.o win_game.o
```

Comment:
Define a variable as follows. The main rule just needs on recipe like what i did.
use the following:
OBJECTS = build/main.o build/credit.o build/game_over.o build/generate_view.o build/is_barrier_hit.o \
 		  build/is_player_hit.o build/is_reach_goal.o build/load_map.o build/print_menu.o \
 		  build/single_player.o build/take_heart.o build/update_arrow.o build/update_barrier.o \
 		  build/update_player_pos.o build/user_manual.o build/win_game.o
 		 
main : $(OBJECTS)
	$(CC) -o main $(OBJECTS) $(LIBFLAGS)

Reply:
That's much better!!!! Thanks for your comments on this.